File: leetcode/1689.py
class Solution:
    def minPartitions(self, n: str) -> int:
        for a in range(9, 0, -1):
            if str(a) in n:
                return a
        return 0

#class Solution:
#    def minPartitions(self, n: str) -> int:
#        return int(max(n))

#class Solution:
#    def minPartitions(self, n: str) -> int:
#        return max(int(c) for c in n)

File: leetcode/1689.rs
impl Solution {
    pub fn min_partitions(n: String) -> i32 {
        n.chars().fold(0, |acc, c| {
            let a: i32 = c.to_digit(10).unwrap() as i32;
            if a > acc { a } else { acc }
        })
    }
}
